### PR TITLE
Add Usage Model

### PR DESCRIPTION
**Pull request by postables**
Allows for controlling account usage.

### CODE REVIEW

File: models/usage.go [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+	"time"
+
+	"github.com/c2h5oh/datasize"
+
+	"github.com/jinzhu/gorm"
+)
+
+// DataUsageTier is a type of usage tier
+// which governs the price per gb ratio
+type DataUsageTier string
```
Comment (bobheadxi) [inline models/usage.go]: very nice use of typed strings 😁

File: models/usage.go [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+	case Light:
+		return 0.22
+	case Plus:
+		return 0.165
+	case Partner:
+		return 0.16
+	default:
+		// this is a catch-all for free tier
+		// free tier users will never encounter a charge call
+		return 9999
```
Comment (bobheadxi) [inline models/usage.go]: `-1` is the standard way to do this i think? doesnt matter that as long as we're careful tho

File: models/usage.go [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+// which governs the price per gb ratio
+type DataUsageTier string
+
+// String returns the value of DataUsageTier as a string
+func (d DataUsageTier) String() string {
+	return string(d)
+}
+
+// PricePerGB returns the price per gb of a usage tier
+// rtc indicates whether or not its an RTC based paymetn
```
Comment (bobheadxi) [inline models/usage.go]: where does this `rtc` variable show up?

File: models/usage.go [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+		IPNSRecordsPublished:    0,
+		PubSubMessagesSent:      0,
+		PrivateNetworkTrialUsed: false,
+		TrialEndTime:            0,
+		Tier:                    tier,
+	}
+	if tier == Free {
+		usage.MonthlyDataLimitGB = datasize.GB.GBytes()
+	} else {
+		usage.MonthlyDataLimitGB = datasize.TB.GBytes()
```
Comment (bobheadxi) [inline models/usage.go]: what's going on here? (one using `GB`, another using `TB`)
also the documentation implies free gets 3gb but it seems to be just 1gb here

File: models/usage.go [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+func (bm *UsageManager) HasStartedPrivateNetworkTrial(username string) (bool, error) {
+	b, err := bm.FindByUserName(username)
+	if err != nil {
+		return false, err
+	}
+	return b.PrivateNetworkTrialUsed, nil
+}
+
+// UpdateTier is used to update the Usage tier associated with an account
+func (bm *UsageManager) UpdateTier(username, tier string) error {
```
Comment (bobheadxi) [inline models/usage.go]:
```suggestion
 func (bm *UsageManager) UpdateTier(username string, tier DataUsageTier) error {
```

File: models/usage.go (hunk 2 above)
Comment (postables) [reply models/usage.go]: Good call

File: models/usage.go (hunk 4 above)
Comment (postables) [reply models/usage.go]: So `datasize.GB.GBytes()` is 1GB represented as GB (you can also do bytes, etc..) while `datasize.TB.GBytes()` is 1TB represented as GB.

Good catch I need to udpate that

File: models/usage.go (hunk 5 above)
Comment (postables) [reply models/usage.go]: Good catch, I also updated the switch/case to use the typed string

File: models/usage.go (hunk 3 above)
Comment (postables) [reply models/usage.go]: Good catch, accidentally left a comment from an earlier implementation, will remove.

File: models/usage.go (hunk 4 above)
Comment (bobheadxi) [reply models/usage.go]: Actually to make it clear, maybe declare these as package-level constants? just for clarity

Comment (postables) [reply models/usage.go]: Done!

**Pull request merged**